### DISCUSSION TITLE
Proper way to handle an undefined `loader` or `action`

### DISCUSSION THREAD

**Discussion by Jotenges** [remix-run/remix#7984] [Q&A]
Hi, when I have a route that only defines an `action` e.g. an api only route, how do properly respond with a 404 to GET requests?
I could define a loader that returns the 404, but I was wondering if there's also a way to override the default error raising logic to just respond with a 404 when an `action` only route received a GET. The only way I see of doing this is by performing a regex match on the error message in `handleError` in `entry.server` and returning a 404 if it matches `but did not provide a `loader` for route`.

**Comment by kiliman** [comment]
That's an edge case that Remix doesn't handle. I would just export a loader that returns 404.

**Reply by thewh1teagle** [reply]
@kiliman 
So it's not possible to detect that handleError is because of non existent loader / action?
